plugin.audio.somafm/resources/lib/addon.py

```diff
@@ -0,0 +1,240 @@
+import os
+import shutil
+import sys
+import time
+import urllib.parse
+import urllib.request
+import xml.etree.ElementTree as ET
+
+import xbmc
+import xbmcaddon
+import xbmcgui
+import xbmcplugin
+import xbmcvfs
+from xbmcgui import ListItem
+from xbmcplugin import SORT_METHOD_GENRE, SORT_METHOD_LISTENERS, SORT_METHOD_UNSORTED
+
+from resources.lib.channel import Channel
+
+CHANNELS_FILE_NAME = "channels.xml"
+
+__addon__ = "SomaFM"
+__addonid__ = "plugin.audio.somafm"
+__version__ = "2.0.0"
+
+__ms_per_day__ = 24 * 60 * 60 * 1000
+
+
+def log(msg):
+    xbmc.log(str(msg), level=xbmc.LOGDEBUG)
+
+
+log(sys.argv)
+
+rootURL = "https://somafm.com/"
+tempdir = xbmcvfs.translatePath("special://home/userdata/addon_data/%s" % __addonid__)
```

Comment:
This will be down automatically by Kodi I think. No need to do this manually?

Reply:
Kodi will only create the directory if the settings are changed, otherwise the addon is using the userdata directory to store the channels. Let me know if there is a better place for caching channel data.